### PR TITLE
[zstd] Fix clang-cl flags

### PR DESCRIPTION
**Pull request by Neumann-A**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] ~~SHA512s are updated for each updated download~~
- [x] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [x] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/zstd/fix-clang-cl.diff [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+index 0265349fb..fa92ce44f 100644
+--- a/build/cmake/CMakeModules/AddZstdCompilationFlags.cmake
++++ b/build/cmake/CMakeModules/AddZstdCompilationFlags.cmake
+@@ -54,12 +54,22 @@ macro(ADD_ZSTD_COMPILATION_FLAGS)
+         endif ()
+         # Add noexecstack flags
+         # LDFLAGS
+-        if(NOT EMSCRIPTEN)
+-        EnableCompilerFlag("-z noexecstack" false false true)
+-        endif()
```
[the remark this answers is not in the document]
Comment (Neumann-A) [reply ports/zstd/fix-clang-cl.diff:11]: See line 24

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/zstd/fix-clang-cl.diff:11]: I know that it is not gone. But I dislike patching the same spot twice.

**Comment by Neumann-A** [comment]
@FrankXie05 Is there a reason this is not getting reviewed ?

**Review by FrankXie05 — DISMISSED** [review]

**Review by BillyONeal — APPROVED** [review]

File: ports/zstd/fix-clang-cl.diff (hunk 1 above)
[the remark this answers is not in the document]
Comment (BillyONeal) [reply ports/zstd/fix-clang-cl.diff]: I merged the patches.

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**